### ISSUE TITLE
Add an instanceNodes array to the vaadin-overlay

### ISSUE THREAD

**Issue by vicsstar** [vaadin/vaadin-overlay#51]
So developers can get access to child components in a dialog/overlay, it'd make sense to expose an array of instanceNodes.

**Comment by tomivirkki** [comment]
Let's implement this as a getter for the stamped template content: `get instanceNodes() { ... }`. If the template hasn’t been stamped when the property is accessed, the getter would do the stamping.

**Comment by manolo** [comment]
afaik `get instanceNodes() { ... }` will not work in flow since elements used in server-side need to be created by the framework in order to have a valid mapping between the browser element and its equivalent in server side, hence it's an unsupported signature.

What do @Legioth or @gilberto-torrezan think, is this enough for making it work out-of-the-box in flow, or we still need to maintain an overlay-flow handwritten component?

**Comment by gilberto-torrezan** [comment]
Currently, the generated components for Flow only work out of the box with some layout/container element if the element provides a `<slot>` (so JS calls to `element.appendChild(node)` works).

Anything different than that needs some special code on the Flow side.

Now specifically about the Dialog case - our `<flow-component-renderer>` custom webcomponent fixed the problem for us. Java developers can use the `vaadin-dialog` the same way they do with `paper-dialog` - by adding and removing components. It was implemented as part of this ticket: https://github.com/vaadin/vaadin-dialog-flow/issues/13

(The code of the `flow-component-renderer` can be found here: https://github.com/vaadin/flow/blob/master/flow-server/src/main/resources/META-INF/resources/frontend/flow-component-renderer.html )

In summary: helper methods are always welcome of course, but without having a `slot` in the webcomponent, Flow will probably just continue to use the renderer internally, since it is the same strategy used for all the other components that uses `<template>`.

**Comment by tomivirkki** [comment]
Just to clarify, this feature is not designed for Flow but for non-Polymer users in general.

**Comment by manolo** [comment]
ok, so the method is useless in flow.
The only solution right now is through the component-renderer by having a hand-written java wrapper for vaadin-dialog,  thanks for the clarification @gilberto-torrezan

**Comment by tomivirkki** [comment]
Closing in favor of https://github.com/vaadin/components-team-tasks/issues/365
